app/jobs/set_participant_completion_date_job.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# This is a one off job to update all of the induction completion dates from DQT for
+# all of the 2021/2022 ECTs that do not currently have them.
+# It runs in batches of 200 as we have a 300 lookup per minute limit on the API
+# This could get adapted later to be a regular running process with a bit more logic
```

Comment:
One off as in it needs to run small batches multiple times to avoid rate limiting but will be removed/refactored/disabled once it's had chance to complete (maybe one or two nights as it was for the induction start dates). There will need to be a different process with maybe different query logic to run something periodically to update ECTs later.